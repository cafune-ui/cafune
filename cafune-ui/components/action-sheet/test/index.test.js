import ActionSheet from '../';
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

describe('ActionSheet', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <ActionSheet
        title="test title"
        horizon={{
          wrap: false,
          list: [
            {
              name: 'hi',
              subName: 'hi sub',
              icon:
                'https://image.uisdc.com/wp-content/uploads/2018/07/uiii-ps-icon.png',
              badge: true,
              action() {
                console.log('ps hi');
              }
            },
            {
              name: 'hi2',
              subName: 'hi2 sub',
              icon: 'search',
              badge: '2',
              action: '/'
            },
            {
              name: 'hi2',
              subName: 'hi2 sub',
              icon: 'search',
              badge: '2'
            },
            {
              name: 'hi2',
              subName: 'hi2 sub',
              icon: 'search',
              badge: '2'
            },
            {
              name: 'hi2',
              subName: 'hi2 sub',
              icon: 'search',
              badge: '2'
            }
          ]
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
