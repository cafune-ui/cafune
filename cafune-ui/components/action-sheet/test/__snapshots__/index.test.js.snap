// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "horizon": Object {
        "list": Array [
          Object {
            "action": [Function],
            "badge": true,
            "icon": "https://image.uisdc.com/wp-content/uploads/2018/07/uiii-ps-icon.png",
            "name": "hi",
            "subName": "hi sub",
          },
          Object {
            "action": "/",
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
        ],
        "wrap": false,
      },
      "title": "test title",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
